File: lib/profil_kontrol.dart
import 'package:supabase_flutter/supabase_flutter.dart';

/// Web `app/lib/profil.ts` ile aynı zorunlu profil alanları.
const profilAlanlari = ['ad', 'soyad', 'telefon', 'adres', 'iletisim_eposta'];

/// Kullanıcının zorunlu profil alanları (ad, soyad, telefon, adres, e-posta) dolu mu?
bool profilTamMi(User? user) {
  if (user == null) return false;
  final m = user.userMetadata ?? const {};
  return profilAlanlari.every((alan) {
    final v = m[alan];
    return v is String && v.trim().isNotEmpty;
  });
}

File: src/lib/profilKontrol.ts
import type { User } from '@supabase/supabase-js';

/** Web `app/lib/profil.ts` ile aynı zorunlu profil alanları. */
export const profilAlanlari = ['ad', 'soyad', 'telefon', 'adres', 'iletisim_eposta'] as const;

/** Kullanıcının zorunlu profil alanları (ad, soyad, telefon, adres, e-posta) dolu mu? */
export function profilTamMi(user: User | null | undefined): boolean {
  if (!user) return false;
  const m = (user.user_metadata ?? {}) as Record<string, unknown>;
  return profilAlanlari.every((alan) => {
    const v = m[alan];
    return typeof v === 'string' && v.trim().length > 0;
  });
}
